### PR TITLE
lb: Use more neutral terminology

### PR DESCRIPTION
**Pull request by brb**
This PR replaces `svc master` by `svc frontend` and `slave` by `backend slot`.

### CODE REVIEW

**Comment by brb** [comment]
test-me-please

File: pkg/maps/lbmap/lbmap.go [hunk 1]
```diff
@@ -72,7 +72,7 @@ func (*LBBPFMap) UpsertService(
 		}
 		svcVal.SetBackendID(loadbalancer.BackendID(backendID))
 		svcVal.SetRevNat(int(svcID))
-		svcKey.SetSlave(slot) // TODO(brb) Rename to SetSlot
+		svcKey.SetEndpoint(slot) // TODO(brb) Rename to SetSlot
```
Comment (tklauser) [inline pkg/maps/lbmap/lbmap.go]: Remove the TODO? Or rename it to `SetSlot` as indicated?

Comment (qmonnet) [inline pkg/maps/lbmap/lbmap.go]: Should you take that chance to implement your `TODO` instead for this one?
[Edit] Oh, Tobias beat me on that one, sorry.

**Review by qmonnet — APPROVED** [review]
Looks great, thanks for that.

**Review by tklauser — APPROVED** [review]

**Comment by brb** [comment]
test-me-please

File: bpf/lib/common.h [hunk 2]
```diff
@@ -572,7 +572,7 @@ struct ct_entry {
 struct lb6_key {
 	union v6addr address;	/* Service virtual IPv6 address */
 	__be16 dport;		/* L4 port filter, if unset, all ports apply */
-	__u16 slave;		/* Backend iterator, 0 indicates the master service */
+	__u16 endpoint;		/* Backend iterator, 0 indicates the primary service */
```
Comment (joestringer) [inline bpf/lib/common.h]: I note that here we call it `endpoint` (which admittedly is consistent with k8s terminology, but also collides with long-held Cilium terminology for the container network termination point), yet in the comment we clarify it by calling it the backend.

Did you consider frontend/backend terminology?

Comment (borkmann) [reply bpf/lib/common.h]: frontend/backend would actually be more clear, or, short `fe`/`be` if vars become too long.

Comment (aanm) [reply bpf/lib/common.h]: yeah I thought about the same the first time I saw this PR but I couldn't think on an alternative, `fe/be` seems a good approach to me as well.

**Review by aanm — CHANGES_REQUESTED** [review]

File: bpf/lib/common.h (hunk 2 above)
Comment (brb) [reply bpf/lib/common.h]: Yep, frontend/backend makes sense. However, the particular field at the line 575 points to a "service slot" (`lb6_service`) which contains a backend_id pointing to a real backend. This is the main reason why I didn't want to use `backend` here. Maybe we should rename `slave` to `backend_slot` or `backend_no`?

Comment (joestringer) [reply bpf/lib/common.h]: if it contains a `backend_id`, is there anything stopping you from calling it `backend_id`? :-)

id/slot/no, any of these sound reasonable.

Comment (brb) [reply bpf/lib/common.h]: Unfortunately, it doesn't contain an ID: `lb6_key{.slave = 1 ... lb6_service[lb6_key{.slave = 0}].count} => lb6_service.backend_id => lb6_backend`. As you can see, `slave` refers to a "backend slot index" ;-) OK, I think I'm going to rename to `backend_slot`, and add a short comment about relations.

**Comment by brb** [comment]
test-me-please

**Comment by brb** [comment]
test-me-please

**Comment by brb** [comment]
The `checkpatch.pl` failure looks harmless:
```
lb: Use more neutral terminology
=========================================================
WARNING:DEPRECATED_TERM: Use of 'master' is deprecated, please 'frontend', instead.
#6: 
This commit replaces "svc master" by "svc frontend" and "slave" by

WARNING:DEPRECATED_TERM: Use of 'slave' is deprecated, please 'backend', instead.
#6: 
This commit replaces "svc master" by "svc frontend" and "slave" by


NOTE: For some of the reported defects, checkpatch may be able to
      mechanically convert to the typical style using --fix or --fix-inplace.

"[PATCH] lb: Use more neutral terminology" has style problems, please review.

NOTE: Ignored message types: C99_COMMENT_TOLERANCE COMPLEX_MACRO CONSTANT_CONVERSION CONST_STRUCT EMAIL_SUBJECT JIFFIES_COMPARISON LEADING_SPACE LONG_LINE_COMMENT MACRO_WITH_FLOW_CONTROL MULTISTATEMENT_MACRO_USE_DO_WHILE PRINTK_WITHOUT_KERN_LEVEL TRAILING_SEMICOLON TRAILING_STATEMENTS VOLATILE

NOTE: If any of the errors are false positives, please report
      them to the maintainer, see CHECKPATCH in MAINTAINERS.
All done
make: *** [checkpatch] Error 1
Makefile.bpf:47: recipe for target 'checkpatch' failed
make: Leaving directory '/home/runner/work/cilium/cilium/bpf'
##[error]Process completed with exit code 2.
```

**Comment by brb** [comment]
test-me-please

**Comment by borkmann** [comment]
I've added the 1.8 backport flag to avoid merge conflicts from this cleanup with future fixes.

**Pull request merged**